### PR TITLE
User management revamp

### PR DESCRIPTION
**Pull request by deye9**

### CODE REVIEW

File: services/user-management/src/main/java/com/foodie/user/model/User.java [hunk 1]
```diff
@@ -28,57 +24,21 @@
 @AllArgsConstructor
 @Entity
 @Table(name = "users")
-public class User extends FoodieBaseEntity implements UserDetails {
+public class User extends FoodieBaseEntity {
```
Comment (sfagade) [inline services/user-management/src/main/java/com/foodie/user/model/User.java:27]: You should implement Serializable in your model classes

File: services/user-management/src/main/java/com/foodie/user/service/UserService.java
Comment (sfagade) [inline services/user-management/src/main/java/com/foodie/user/service/UserService.java:1]: You have unused `imports` in this file

File: services/user-management/src/main/java/com/foodie/user/service/UserService.java [hunk 2]
```diff
@@ -1,53 +1,71 @@
...
+        if (connectedUser instanceof Principal) {
+            retrievedUser = ((FoodieUserDetails) ((UsernamePasswordAuthenticationToken) connectedUser).getPrincipal()).getUser();
+        } else if (connectedUser instanceof UUID) {
+            retrievedUser =  findByIdAndDeletedAtIsNull((UUID) connectedUser).orElse(null);
+        } else {
+            throw new IllegalArgumentException("Invalid connectedUser type");
+        }
+
+        if (retrievedUser == null || !retrievedUser.getIsActive()) {
+            throw new RuntimeException("Requested user not found or not active");
```
Comment (sfagade) [inline services/user-management/src/main/java/com/foodie/user/service/UserService.java:44]: Not a good idea to throw `RuntimeException` use a more specific exception or create a custom exception class

File: services/user-management/src/main/java/com/foodie/user/utils/FoodieUserDetails.java [hunk 3]
```diff
@@ -0,0 +1,63 @@
...
+
+import com.foodie.user.model.User;
+import org.springframework.security.core.GrantedAuthority;
+import org.springframework.security.core.userdetails.UserDetails;
+import java.util.Collection;
+import java.util.UUID;
+
+public class FoodieUserDetails implements UserDetails {
+
+    private User user;
```
Comment (sfagade) [inline services/user-management/src/main/java/com/foodie/user/utils/FoodieUserDetails.java:11]: Make final

File: services/user-management/src/main/java/com/foodie/user/utils/FoodieUserDetails.java [hunk 4]
```diff
@@ -0,0 +1,63 @@
...
+
+    public FoodieUserDetails(User user) {
+        this.user = user;
+    }
+
+    public User getUser() {
```
Comment (sfagade) [inline services/user-management/src/main/java/com/foodie/user/utils/FoodieUserDetails.java:17]: Use lombok?

File: services/user-management/src/test/java/com/foodie/user/TestSetupPostgres.java
Comment (sfagade) [inline services/user-management/src/test/java/com/foodie/user/TestSetupPostgres.java:1]: I would have probably used H2 db for tests, that way it's cost-effective and easy to teardown

File: services/user-management/src/test/java/com/foodie/user/controllers/UserControllerTest.java [hunk 5]
```diff
@@ -0,0 +1,269 @@
...
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Test;
+import org.springframework.http.HttpStatus;
+import org.springframework.http.ResponseEntity;
+
+import com.foodie.FoodieBaseResponse;
+import com.foodie.user.model.User;
+import com.foodie.user.service.UserService;
+
+public class UserControllerTest {
```
Comment (sfagade) [inline services/user-management/src/test/java/com/foodie/user/controllers/UserControllerTest.java:26]: class doesn't need to be public

**Review by sfagade — APPROVED** [review]
See comments. Who made you change your architecture ?

File: services/user-management/src/main/java/com/foodie/user/model/User.java (hunk 1 above)
Comment (deye9) [reply services/user-management/src/main/java/com/foodie/user/model/User.java:27]: Will look into it

File: services/user-management/src/main/java/com/foodie/user/service/UserService.java
Comment (deye9) [reply services/user-management/src/main/java/com/foodie/user/service/UserService.java:1]: Removed

File: services/user-management/src/main/java/com/foodie/user/service/UserService.java (hunk 2 above)
Comment (deye9) [reply services/user-management/src/main/java/com/foodie/user/service/UserService.java:44]: Will do so

File: services/user-management/src/main/java/com/foodie/user/utils/FoodieUserDetails.java (hunk 4 above)
Comment (deye9) [reply services/user-management/src/main/java/com/foodie/user/utils/FoodieUserDetails.java:17]: Did not use it cause of this custom detail

@Override
    public String getUsername() {
        return user.getEmail();
    }

File: services/user-management/src/test/java/com/foodie/user/TestSetupPostgres.java
Comment (deye9) [reply services/user-management/src/test/java/com/foodie/user/TestSetupPostgres.java:1]: Might move to it later

**Comment by deye9** [comment]
Architecture change was bound to happen. I was still ramping up on Java and needed time

File: services/user-management/src/main/java/com/foodie/user/utils/FoodieUserDetails.java (hunk 4 above)
Comment (sfagade) [reply services/user-management/src/main/java/com/foodie/user/utils/FoodieUserDetails.java:17]: You can still have it in the class. Your custom details will override the ones Lombok created

**Pull request merged**